runtime/runtime.go

```diff
@@ -372,9 +398,52 @@ func (rt *Runtime) StartREPL(ctx context.Context) {
 		}
 	}
 
+	if !rt.Params.Incognito {
+		go rt.checkOPAUpdate(ctx, nil)
+	}
 	repl.Loop(ctx)
 }
 
+func (rt *Runtime) checkOPAUpdate(ctx context.Context, done chan struct{}) {
+	resp, err := rt.reporter.SendReport(ctx)
+	if err != nil {
+		return
+	}
+
+	banner := resp.Pretty()
+	if banner != "" {
+		fmt.Fprintln(rt.Params.Output, banner)
+		fmt.Fprintf(rt.Params.Output, "> ")
+	}
+	return
+}
+
+func (rt *Runtime) checkOPAUpdateLoop(ctx context.Context, uploadInterval int64, done chan struct{}) {
+	ticker := time.NewTicker(time.Duration(int64(time.Second) * uploadInterval))
+
+	for {
+		select {
+		case <-ticker.C:
+			resp, err := rt.reporter.SendReport(ctx)
+			if err != nil {
+				logrus.WithField("err", err).Debug("Unable to send OPA version report.")
+			} else {
+				if resp.Latest.Download == "" || resp.Latest.ReleaseNotes == "" {
+					logrus.Debug("OPA version is up-to-date !")
+				} else {
+					logrus.WithFields(logrus.Fields{
+						"download_opa":  resp.Latest.Download,
+						"release_notes": resp.Latest.ReleaseNotes,
+					}).Infof("A newer version of OPA is available for download ! Current version %v", version.Version)
+				}
```

Comment:
Nit; How about:

```
OPA is up-to-date.
```

and

```
OPA is out-of-date.
```

Both can log the current version and latest version. BTW, I notice the response only contains URLs for download/changelog. Should we include for the version as well?

Reply:
Are you referring to  https://github.com/open-policy-agent/opa/tags ?

We can update `A newer version of OPA is available for download !` with `OPA is out-of-date.`  and log the current version in both up-to-date and out-of-date cases. 

In the case where the running OPA version is already greater than the released version, the telemetry server does not send any information in the response. IMO it should be fine to only show the current version if the running OPA version is already equal or more than the released.

Reply:
Updated logs:
```
{"level":"debug","msg":"OPA is up-to-date. Current version 0.20.0","time":"2020-05-18T23:43:56-07:00"}
```

```
{"download_opa":"https://openpolicyagent.org/downloads/v0.22.0/opa_darwin_amd64","level":"info","msg":"OPA is out-of-date. Current version 0.20.0. Latest version v0.22.0","release_notes":"https://github.com/open-policy-agent/opa/releases/tag/v0.22.0","time":"2020-05-18T23:45:27-07:00"}
```

Reply:
Looks better. Take the "current version XYZ" out of the message though and move it into data. The message should be "OPA is out of date" and the data should be contained inside separate fields.

Reply:
Cool. Updated logs:

```
{"current_version":"0.20.0","level":"debug","msg":"OPA is up-to-date.","time":"2020-05-19T13:15:33-07:00"}
```

```
{"current_version":"0.20.0","download_opa":"https://openpolicyagent.org/downloads/v0.22.0/opa_darwin_amd64","latest_version":"v0.22.0","level":"info","msg":"OPA is out-of-date.","release_notes":"https://github.com/open-policy-agent/opa/releases/tag/v0.22.0","time":"2020-05-19T13:15:13-07:00"}
```